includes/class-wc-gateway-stripe.php

```diff
@@ -842,28 +843,37 @@ public function retry_after_error( $response, $order, $retry, $force_save_source
 	 * it up and prepare it for the Stripe PaymentIntents modal to confirm a payment.
 	 *
 	 * @since 4.2
-	 * @param WC_Payment_Gateway[] $gateways A list of all available gateways.
-	 * @return WC_Payment_Gateway[]          Either the same list or an empty one in the right conditions.
 	 */
-	public function prepare_order_pay_page( $gateways ) {
+	public function prepare_order_pay_page() {
 		if ( ! is_wc_endpoint_url( 'order-pay' ) || ! isset( $_GET['wc-stripe-confirmation'] ) ) { // wpcs: csrf ok.
-			return $gateways;
+			return;
 		}
 
 		try {
 			$this->prepare_intent_for_order_pay_page();
 		} catch ( WC_Stripe_Exception $e ) {
 			// Just show the full order pay page if there was a problem preparing the Payment Intent
-			return $gateways;
+			return;
 		}
 
 		add_filter( 'woocommerce_checkout_show_terms', '__return_false' );
 		add_filter( 'woocommerce_pay_order_button_html', '__return_false' );
 		add_filter( 'woocommerce_available_payment_gateways', '__return_empty_array' );
 		add_filter( 'woocommerce_no_available_payment_methods_message', array( $this, 'change_no_available_methods_message' ) );
-		add_action( 'woocommerce_pay_order_after_submit', array( $this, 'render_payment_intent_inputs' ) );
+		add_action( 'woocommerce_pay_order_after_submit', array( $this, 'render_intent_inputs' ) );
+		add_action( 'after_woocommerce_pay', array( $this, 'render_intent_inputs' ), 101 );
```

Comment:
Why this unusual priority? Please add a comment to the code.

Reply:
The "Pay for order" form is replaced with "Change payment method" by WooCommerce Subscriptions, on this hook with priority 100 – does the new comment in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1405/commits/6c9833e0def6bd467b04e80f0d07554530dcf3df make that clear enough?

Reply:
Yes, ty!